Revise Configuration Structure spec (#57)
riscv/configuration-structure: README.md

- schema.asn is the ASN.1 schema used that contains the configuration structure
  format.

> ## Just The Basics

Reviewer: ```suggestion
## Extend The Schema
```

2. If the list is long, divide the decisions into primary, secondary, and
complete decisions. Primary information is only discoverable by reading the
configuration structure, or by running a significant amount of code. Secondary
information is discoverable but not straightforward (e.g. WARL register).
Anything else goes in the complete section.
3. Get ready to make a pull request to
https://github.com/riscv/configuration-structure, e.g. by cloning the repo.
4. Write an ASN.1 description of this information. ASN.1 allows for all kinds of
complexity, but for your purposes simply reading one of extension .asn files
under the schema/ directory should be all you need. For a simple example, look
at zjpm. For a complex example look at debug.
5. Following those examples, make a new file for your extension. Then edit
configuration-structure.asn, adding to the IMPORTS and adding appropriate
reference in the Configuration and Harts section. (You might only need one of
those.)
6. Run `make test` at the top level, and make sure no errors are reported.
7. Make a pull request against the github repo.

If you have any questions during this process, you might find the answers

> further down in this document. Otherwise, please contact us at

Reviewer: ```suggestion
in the spec. Otherwise, please contact us at
```

tech-config@lists.riscv.org.

> ## Goals of Task Group

Reviewer: This section really doesn't belong in the README. README is a basic document to get people pointed in the right direction. I agree that it might not belong in the spec either, but the spec does need some introduction that explains the problem it is trying to address. I'd be inclined to leave it in the spec, but if you don't want that, then how about making a new Background document for this kind of stuff?

Author: ok, we can have a Background document.

a configuration structure, and to display the configuration structure to
people.
. A software tool that translates back and forth between the
machine-readable and human-readable format.
. A specification for a method to discover and access, from M-mode
software, the machine-readable configuration structures.

The configuration structure should be flexible enough that future task
groups won’t feel the need to create another structure used to describe
implementation parameters. Implementation parameters are details that a
RISC-V specification explicitly leaves up to an implementation. This
includes hart-specific details like the kinds of hardware triggers
supported, as well as details that are outside harts such as the
supported abstract debug commands.

The configuration structure can be specified/implemented/configured/extended at
different points in supply chain (IP supplier, SoC integrator,
manufacture/testing SKU fuse programming, board assembly, board programming).
The solution will accommodate hundreds of different types of RISC-V core,
integrated onto thousands of unique SoCs per year, assembled into tens of
thousands of unique boards per year.

> ## Considerations

Reviewer: Same as above.

Author: yes, move this to the Background document.
